common/utils.js

```diff
@@ -492,6 +503,10 @@
         }
 
 
+        // TODO: If we decide to catch errors in this file instead of call site,
+        // then we could wrap each of the returned functions with the catcher fn
+        // so that it's all in one place. Same with the other util functions returned
+        // in the factories/services in this file.
```

Comment:
I think that's what we want to do. Except when we define the wrapper function in here, we don't want to wrap it again.